update.ps1

```diff
@@ -1,201 +1,532 @@
-    "userPrincipalName"
-    "mail"
-)
-
-
-# Start Script
 try {
-    # Formatting Headers and authentication for KPN Lisa Requests
-    $LisaRequest = @{
-        Authentication = "Bearer"
-        Token          = $actionContext.Configuration.AzureAD | Get-LisaAccessToken -AsSecureString
-        ContentType    = "application/json; charset=utf-8"
-        Headers        = @{
-            "Mwp-Api-Version" = "1.0"
+    #region account
+    # Define correlation
+    $correlationField = "Id"
+    $correlationValue = $actionContext.References.Account
+
+    # Define account object
+    $account = [PSCustomObject]$actionContext.Data.PsObject.Copy()
+
+    # Define properties to query
+    $accountPropertiesToQuery = @("id") + $account.PsObject.Properties.Name | Select-Object -Unique
+
+    # Remove properties of account object with null-values
+    $account.PsObject.Properties | ForEach-Object {
+        # Remove properties with null-values
+        if ($_.Value -eq $null) {
+            $account.PsObject.Properties.Remove("$($_.Name)")
         }
     }
-
-    #Get previous account, select only $outputContext.Data.Keys
-    $SplatParams = @{
-        Uri    = "$($actionContext.Configuration.BaseUrl)/Users/$($actionContext.References.Account)"
-        Method = "Get"
+    # Convert the properties of account object containing "TRUE" or "FALSE" to boolean
+    $account = Convert-StringToBoolean $account
+
+    # Define properties to compare for update
+    $accountPropertiesToCompare = $account.PsObject.Properties.Name
+    #endRegion account
+
+    #region Verify account reference
+    $actionMessage = "verifying account reference"
+    
+    if ([string]::IsNullOrEmpty($($actionContext.References.Account))) {
+        throw "The account reference could not be found"
+    }
+    #endregion Verify account reference
+    
+    #region Create access token
+    $actionMessage = "creating access token"
+    
+    $createAccessTokenBody = @{
+        grant_type    = "client_credentials"
+        client_id     = $actionContext.Configuration.EntraIDAppId
+        client_secret = $actionContext.Configuration.EntraIDAppSecret
+        scope         = $actionContext.Configuration.KPNMWPScope
+    }
+    
+    $createAccessTokenSplatParams = @{
+        Uri             = "https://login.microsoftonline.com/$($actionContext.Configuration.EntraIDTenantID)/oauth2/v2.0/token/"
+        Headers         = $headers
+        Method          = "POST"
+        ContentType     = "application/x-www-form-urlencoded"
+        UseBasicParsing = $true
+        Body            = $createAccessTokenBody
+        Verbose         = $false
+        ErrorAction     = "Stop"
+    }
+    
+    $createAccessTokenResonse = Invoke-RestMethod @createAccessTokenSplatParams
+    
+    Write-Verbose "Created access token. Result: $($createAccessTokenResonse | ConvertTo-Json)"
+    #endregion Create access token
+    
+    #region Create headers
+    $actionMessage = "creating headers"
```

Comment:
Deze 'actionMessage' zal je nooit terugzien in de catch omdat de code tot de volgende actionMessage eigenlijk niet fout kan gaan.